backend/models/generic.go

```diff
@@ -1,13 +1,11 @@
 package models
 
 import (
-	"gorm.io/gorm"
 	"time"
 )
 
 type Common struct {
-	Id        uint           `gorm:"primaryKey" json:"id"`
-	CreatedAt time.Time      `json:"created_at"`
-	UpdatedAt time.Time      `json:"updated_at"`
-	DeletedAt gorm.DeletedAt `gorm:"index" json:"deleted_at"`
```

Comment:
I left a previous comment about this, what's the point of it

Reply:
Replaces the generic gorm.model that is the whole point...

Reply:
Does it need the `DeletedAt` field though, this part confuses me because we don't use it.

Reply:
Including the `DeletedAt` field with type `gorm.DeletedAt` changes the behavior of `db.Delete(&object)` to perform a soft delete on that object. Instead of removing the record from the database, gorm will set the `DeletedAt`'s value to the current time and that record will no longer be findable with normal query methods.

I think this feature is normally used for services that have a "recovery" option, like deactivating a social media account. It doesn't seem like we need this functionality for any part of our service, so I think it would be safe to delete just the `DeletedAt` field. What do you think @ABarroso647 ?